tests/scanners/test.base.js

```diff
@@ -13,6 +13,16 @@ class BaseScannerWithContents extends BaseScanner {
 
 describe('Base Scanner Class', function() {
 
+  it('scannerName should throw an error', () => {
+    try {
+      var scanner = new BaseScanner('', 'filename.txt');
+      scanner.scannerName;
+    } catch (err) {
+      assert.instanceOf(err, Error);
+      assert.equal(err.message, 'scannername is not implemented');
```

Comment:
It's weird that this works as the actual error is `scannerName is not implemented` (note the camelCase), and it should definitely be camelCase everywhere so it's easier to scan/grok. Maybe `strictEqual` would do the trick?